api/src/main/java/io/opentelemetry/common/Attributes.java

```diff
@@ -0,0 +1,239 @@
+
+    @Override
+    abstract List<Object> data();
+  }
+
+  /** An {@link Attributes} instance with no attributes. */
+  public static Attributes empty() {
+    return EMPTY;
+  }
+
+  /** An {@link Attributes} instance with a single key-value pair. */
+  public static Attributes of(String key, AttributeValue value) {
+    return sortAndFilterToAttributes(key, value);
+  }
+
+  /**
+   * An {@link Attributes} instance with two key-value pairs. Order of the keys is not preserved.
+   * Duplicate keys will be removed.
+   */
+  public static Attributes of(
+      String key1, AttributeValue value1, String key2, AttributeValue value2) {
+    return sortAndFilterToAttributes(key1, value1, key2, value2);
+  }
+
+  /**
+   * An {@link Attributes} instance with three key-value pairs. Order of the keys is not preserved.
+   * Duplicate keys will be removed.
+   */
+  public static Attributes of(
+      String key1,
+      AttributeValue value1,
+      String key2,
+      AttributeValue value2,
+      String key3,
+      AttributeValue value3) {
+    return sortAndFilterToAttributes(key1, value1, key2, value2, key3, value3);
+  }
+
+  /**
+   * An {@link Attributes} instance with four key-value pairs. Order of the keys is not preserved.
+   * Duplicate keys will be removed.
+   */
+  public static Attributes of(
+      String key1,
+      AttributeValue value1,
+      String key2,
+      AttributeValue value2,
+      String key3,
+      AttributeValue value3,
+      String key4,
+      AttributeValue value4) {
+    return sortAndFilterToAttributes(key1, value1, key2, value2, key3, value3, key4, value4);
+  }
+
+  /**
+   * An {@link Attributes} instance with five key-value pairs. Order of the keys is not preserved.
+   * Duplicate keys will be removed.
+   */
+  public static Attributes of(
+      String key1,
+      AttributeValue value1,
+      String key2,
+      AttributeValue value2,
+      String key3,
+      AttributeValue value3,
+      String key4,
+      AttributeValue value4,
+      String key5,
+      AttributeValue value5) {
+    return sortAndFilterToAttributes(
+        key1, value1,
+        key2, value2,
+        key3, value3,
+        key4, value4,
+        key5, value5);
+  }
+
+  private static Attributes sortAndFilterToAttributes(Object... data) {
+    return new AutoValue_Attributes_ArrayBackedAttributes(sortAndFilter(data));
+  }
+
+  /** Creates a new {@link Builder} instance for creating arbitrary {@link Attributes}. */
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  /**
+   * Enables the creation of an {@link Attributes} instance with an arbitrary number of key-value
+   * pairs.
+   */
+  public static class Builder {
+    private final List<Object> data = new ArrayList<>();
+
+    /** Create the {@link Attributes} from this. */
+    public Attributes build() {
+      return sortAndFilterToAttributes(data.toArray());
+    }
+
+    /**
+     * Add a bare {@link AttributeValue} to this.
```

Comment:
Sets instead of Add throughout (matches the method name and javadoc should generally be present tense, not command tense)

Reply:
done